daml-lf/tests/Exceptions.daml

```diff
@@ -103,6 +103,33 @@ template T
           throw (Ecid cid)
         catch (Ecid cid) -> archive cid
 
+    nonconsuming choice ThrowInHandler : ()
```

Comment:
Three new test cases.
- The first worked anyway
- The second and third were previously broken